lib/reverse_sentence.rb

```diff
@@ -1,6 +1,22 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+ 
+  if my_sentence == nil
+    return nil
+  elsif 
+    my_sentence == ""
+    return ""
+  end
+
+  array = my_sentence.split(/(\S*|\s*)/) # Split by each word and each space
```

Comment:
This is making another array causing O(n) space complexity.  You were asked to do the reverse in place.